lib/pages/chat/events/sending_video_widget.dart

```diff
@@ -0,0 +1,166 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:chewie/chewie.dart';
+import 'package:fluffychat/pages/chat/events/message_content_style.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/linagora_design_flutter.dart';
+import 'package:matrix/matrix.dart';
+import 'package:video_player/video_player.dart';
+
+class SendingVideoWidget extends StatefulWidget {
+
+  final Event event;
+
+  final Uint8List bytes;
+
+  final String filePath;
+
+
+  const SendingVideoWidget({
+    super.key,
+    required this.event,
+    required this.bytes,
+    required this.filePath,
+  });
+
+  @override
+  State<SendingVideoWidget> createState() => _SendingVideoWidgetState();
+}
+
+class _SendingVideoWidgetState extends State<SendingVideoWidget> with AutomaticKeepAliveClientMixin {
+  final ValueNotifier<double> sendingFileProgressNotifier = ValueNotifier(0);
+  ChewieController? chewieController;
+  VideoPlayerController? videoPlayerController;
+
+  bool isPlay = false;
+
+  @override
+  void dispose() {
+    videoPlayerController?.dispose();
+    chewieController?.dispose();
+    super.dispose();
+  }
+
+  Future<void> initVideoController() async {
+    videoPlayerController = VideoPlayerController.file(
+      File(widget.filePath),
+    );
+    await videoPlayerController?.initialize();
+    chewieController = ChewieController(
+      videoPlayerController: videoPlayerController!,
+      autoPlay: false,
+      showControlsOnInitialize: false,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (widget.event.status == EventStatus.sent || widget.event.status == EventStatus.synced) {
+      sendingFileProgressNotifier.value = 1;
```

Comment:
don't update anything in build func
use listener in controller

Reply:
The thing is `event` can come from anywhere anytime, so listening for new events in the controller is needed to parse a lot to recognize the event for image sending. Instead of doing that, When a new event is coming to this widget, we update the sendingFileProgress, if it's the same, nothing rebuild.